cmd/contour/certgen.go

```diff
@@ -137,34 +138,47 @@ func OutputCerts(config *certgenConfig, kubeclient *kubernetes.Clientset, certs
 		case "compact":
 			secrets = certgen.AsSecrets(config.Namespace, certs)
 		default:
-			check(fmt.Errorf("unsupported Secrets format %q", config.Format))
+			return fmt.Errorf("unsupported Secrets format %q", config.Format)
 		}
 	}
 
 	if config.OutputPEM {
 		fmt.Printf("Writing certificates to PEM files in %s/\n", config.OutputDir)
-		check(certgen.WriteCertsPEM(config.OutputDir, certs, force))
+		if err := certgen.WriteCertsPEM(config.OutputDir, certs, force); err != nil {
+			return fmt.Errorf("Writing certificates to PEM files in %s/\n", config.OutputDir)
+		}
 	}
 
 	if config.OutputYAML {
 		fmt.Printf("Writing %q format Secrets to YAML files in %s/\n", config.Format, config.OutputDir)
-		check(certgen.WriteSecretsYAML(config.OutputDir, secrets, force))
+		if err := certgen.WriteSecretsYAML(config.OutputDir, secrets, force); err != nil {
+			return fmt.Errorf("failer to write %q format Secrets to YAML files in %s/\n", config.Format, config.OutputDir)
+		}
 	}
 
 	if config.OutputKube {
 		fmt.Printf("Writing %q format Secrets to namespace %q\n", config.Format, config.Namespace)
-		check(certgen.WriteSecretsKube(kubeclient, secrets, force))
+		if err := certgen.WriteSecretsKube(kubeclient, secrets, force); err != nil {
+			return fmt.Errorf("failed to write  %q format Secrets to namespace %q\n", config.Format, config.Namespace)
+		}
 	}
+	return nil
 }
 
-func doCertgen(config *certgenConfig) {
+func doCertgen(config *certgenConfig, log logrus.FieldLogger) {
+
 	generatedCerts, err := GenerateCerts(config)
-	check(err)
+	if err != nil {
+		log.WithError(err).Fatal("failed to generate certs")
```

Comment:
Nit: we should avoid abbreviations in user-visible messages

```Go
log.WithError(err).Fatal("failed to generate certificates")
```

Reply:
Done.